src/Catalyst.Core.Modules.Dfs.Tests/IntegrationTests/CoreApi/BlockApiTest.cs

```diff
@@ -0,0 +1,314 @@
+
+            var data = ipfs.BlockApi.GetAsync(cid).Result;
+            Assert.Equal(blob.Length, data.Size);
+            Assert.Equal(blob, data.DataBytes);
+        }
+
+        [Fact]
+        public void Put_Stream_Hash()
+        {
+            var cid = ipfs.BlockApi.PutAsync(new MemoryStream(blob), "raw", "sha2-512").Result;
+            Assert.Equal(
+                "bafkrgqelljziv4qfg5mefz36m2y3h6voaralnw6lwb4f53xcnrf4mlsykkn7vt6eno547tw5ygcz62kxrle45wnbmpbofo5tvu57jvuaf7k7e",
+                (string) cid);
+
+            var data = ipfs.BlockApi.GetAsync(cid).Result;
+            Assert.Equal(blob.Length, data.Size);
+            Assert.Equal(blob, data.DataBytes);
+        }
+
+        [Fact]
+        public void Get()
+        {
+            var _ = ipfs.BlockApi.PutAsync(blob).Result;
+            var block = ipfs.BlockApi.GetAsync(id).Result;
+            Assert.Equal(id, (string) block.Id);
+            Assert.Equal(blob, block.DataBytes);
+            var blob1 = new byte[blob.Length];
+            block.DataStream.Read(blob1, 0, blob1.Length);
+            Assert.Equal(blob, blob1);
+        }
+
+        [Fact]
+        public void Stat()
+        {
+            var _ = ipfs.BlockApi.PutAsync(blob).Result;
+            var info = ipfs.BlockApi.StatAsync(id).Result;
+            Assert.Equal(id, (string) info.Id);
+            Assert.Equal(5, info.Size);
+        }
+
+        [Fact]
+        public async Task Stat_Inline_CID()
+        {
+            var cts = new CancellationTokenSource(300);
+            var cid = new Cid
+            {
+                ContentType = "raw",
+                Hash = MultiHash.ComputeHash(blob, "identity")
+            };
+            var info = await ipfs.BlockApi.StatAsync(cid, cts.Token);
+            Assert.Equal(cid.Encode(), (string) info.Id);
+            Assert.Equal(5, info.Size);
+        }
+
+        [Fact]
+        public async Task Stat_Unknown()
+        {
+            var cid = "QmPv52ekjS75L4JmHpXVeuJ5uX2ecSfSZo88NSyxwA3rFF";
+            var block = await ipfs.BlockApi.StatAsync(cid);
+            Assert.Null(block);
+        }
+
+        [Fact]
+        public async Task Remove()
+        {
+            var _ = ipfs.BlockApi.PutAsync(blob).Result;
+            var cid = await ipfs.BlockApi.RemoveAsync(id);
+            Assert.Equal(id, (string) cid);
+        }
+
+        [Fact]
+        public async Task Remove_Inline_CID()
+        {
+            var cid = new Cid
+            {
+                ContentType = "raw",
+                Hash = MultiHash.ComputeHash(blob, "identity")
+            };
+            var removedCid = await ipfs.BlockApi.RemoveAsync(cid);
+            Assert.Equal(cid.Encode(), removedCid.Encode());
+        }
+
+        [Fact]
+        public void Remove_Unknown()
+        {
+            ExceptionAssert.Throws<Exception>(() =>
+            {
+                var _ = ipfs.BlockApi.RemoveAsync("QmPv52ekjS75L4JmHpXVeuJ5uX2ecSfSZo88NSyxwA3rFF").Result;
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Remove the unused local variable '_'.](https://app.codacy.com/manual/catalyst-network/Catalyst.Framework/pullRequest?prid=4832783)